packages/typespec-test/test/batch_modular/generated/typespec-ts/src/util/serializeUtil.ts

```diff
@@ -0,0 +1,3372 @@
+        ? o["certificateReferences"]
+        : o["certificateReferences"].map((e) => MISSING_SERIALIZER(e)),
+    startTask:
+      o["startTask"] === undefined
+        ? o["startTask"]
+        : MISSING_SERIALIZER(o["startTask"]),
+  };
+}
+
+export function serializeBatchPoolCreateOptions(
+  o: BatchPoolCreateOptions,
+): RestBatchPoolCreateOptions {
+  return {
+    targetNodeCommunicationMode: o["targetNodeCommunicationMode"],
+    mountConfiguration:
+      o["mountConfiguration"] === undefined
+        ? o["mountConfiguration"]
+        : o["mountConfiguration"].map((e) => MISSING_SERIALIZER(e)),
+    metadata:
+      o["metadata"] === undefined
+        ? o["metadata"]
+        : o["metadata"].map((e) => MISSING_SERIALIZER(e)),
+    userAccounts:
+      o["userAccounts"] === undefined
+        ? o["userAccounts"]
+        : o["userAccounts"].map((e) => MISSING_SERIALIZER(e)),
+    taskSchedulingPolicy:
+      o["taskSchedulingPolicy"] === undefined
+        ? o["taskSchedulingPolicy"]
+        : MISSING_SERIALIZER(o["taskSchedulingPolicy"]),
+    taskSlotsPerNode: o["taskSlotsPerNode"],
+    applicationLicenses: o["applicationLicenses"],
+    applicationPackageReferences:
+      o["applicationPackageReferences"] === undefined
+        ? o["applicationPackageReferences"]
+        : o["applicationPackageReferences"].map((e) => MISSING_SERIALIZER(e)),
+    certificateReferences:
+      o["certificateReferences"] === undefined
+        ? o["certificateReferences"]
+        : o["certificateReferences"].map((e) => MISSING_SERIALIZER(e)),
+    startTask:
+      o["startTask"] === undefined
+        ? o["startTask"]
+        : MISSING_SERIALIZER(o["startTask"]),
+    networkConfiguration:
+      o["networkConfiguration"] === undefined
+        ? o["networkConfiguration"]
+        : MISSING_SERIALIZER(o["networkConfiguration"]),
+    enableInterNodeCommunication: o["enableInterNodeCommunication"],
+    autoScaleEvaluationInterval:
+      o["autoScaleEvaluationInterval"] === undefined
+        ? o["autoScaleEvaluationInterval"]
+        : FIXME,
+    autoScaleFormula: o["autoScaleFormula"],
+    enableAutoScale: o["enableAutoScale"],
+    targetLowPriorityNodes: o["targetLowPriorityNodes"],
+    targetDedicatedNodes: o["targetDedicatedNodes"],
+    resizeTimeout:
+      o["resizeTimeout"] === undefined ? o["resizeTimeout"] : FIXME,
+    virtualMachineConfiguration:
+      o["virtualMachineConfiguration"] === undefined
+        ? o["virtualMachineConfiguration"]
+        : MISSING_SERIALIZER(o["virtualMachineConfiguration"]),
+    cloudServiceConfiguration:
+      o["cloudServiceConfiguration"] === undefined
+        ? o["cloudServiceConfiguration"]
+        : MISSING_SERIALIZER(o["cloudServiceConfiguration"]),
+    vmSize: o["vmSize"],
+    displayName: o["displayName"],
+    id: o["id"],
+  };
+}
+
+export function serializeBatchJobCreateOptions(
+  o: BatchJobCreateOptions,
+): RestBatchJobCreateOptions {
+  return {
```

Comment:
I don't know if you remember when we try to support dictionary type, we found that this kind of inline assignment way of serialization and deserialization, wouldn't work if the dictionary value type is complex data type? I can't find the link now.